src/nimlsp.nim

```diff
@@ -145,9 +147,20 @@ proc getProjectFile(file: string): string =
       fileExists(path / current.addFileExt(".nims"))) and certainty <= Cfg:
       result = path / current.addFileExt(".nim")
       certainty = Cfg
-    if fileExists(path / current.addFileExt(".nimble")) and certainty <= Nimble:
-      # Read the .nimble file and find the project file
-      discard
+    for file in walkDirRec(path):
```

Comment:
This would walk down into subdirectories, and since the outer loop goes all the way out to system root it will potentially scan the entire disk for a `.nimble` file. Why exactly is this added? The `.nimble` file should be found in a directory above the current source file, so it shouldn't be necesarry to do this step.

Reply:
walkDir seems good for this.

Reply:
It's not. The loop it is in goes up the directory tree until it hits the root folder of your system. `walkDirRec` goes down into every file and sub-folder below it. So if you don't have a .nimble file anywhere on your system this will search your entire hard-drive for one. And if you have one in a completely unrelated project this will find that and use it as the project file.

Reply:
ok, then I think I need store project root when initial, compare to that.

Reply:
No, just check only the folders it already passes through, without the recursion.